### DISCUSSION TITLE
Dismissing an ActionGroup Menu in Action

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#5212] [Help] [answered]
Is there a way to dismiss/close the ActionGroup menu after clicking an Action within that ActionGroup? After I click my Action, the ActionGroup menu stays visible.

This is my code:

```php
Pages\Actions\ActionGroup::make([
    Pages\Actions\Action::make('checkin')
        ->label('Check In')
        ->icon('heroicon-o-clipboard-check')
        ->color('secondary')
        ->action(function (): void {
            $this->record->update(['arrived_at' => now()]);

            Notification::make()
                ->title('The registrant has been checked in.')
                ->success()
                ->send();
        }),
}),
```

**Comment by intrepidws** [comment, marked as answer]
I was able to add this to the Action to get this to work:

```php
->extraAttributes(['x-on:click' => 'toggle'])
```

**Comment by intrepidws** [comment]
I had this same issue today and found my own issue - there has to be a better way to accomplish this? It feels like it should happen automatically honestly. Anyone have any ideas?

**Reply by bayprogrammer** [reply]
Does chaining a call to `->close()` do the trick? E.g.:

```php
Pages\Actions\ActionGroup::make([
    Pages\Actions\Action::make('checkin')
        ->label('Check In')
        ->icon('heroicon-o-clipboard-check')
        ->color('secondary')
        ->action(function (): void {
            $this->record->update(['arrived_at' => now()]);

            Notification::make()
                ->title('The registrant has been checked in.')
                ->success()
                ->send();
        })
        ->close(),  // <-- add this
]),
```

**Reply by intrepidws** [reply]
That method doesn't exist on the `Filament\Pages\Actions\Action` class.

**Reply by bayprogrammer** [reply]
Ah, bummer. I tested with `Filament\Actions\Action` and `Filament\Tables\Actions\Action`, but not `Filament\Pages\Actions\Action`.

**Comment by hairulloh-sukur** [comment]
I usually do it this way:
```php
                ->hidden(
                    fn() => ($this->record->order_status == 'new') ? false : true
                ),
```
This is reactive and persistent even when the page is refreshed.

**Reply by intrepidws** [reply]
My issue isn't whether or not the `Action` itself should be visible, it's the fact that the `ActionGroup` remains visible after clicking the action button.
